### PR TITLE
Make models

### PR DESCRIPTION
**Pull request by deependra-default**

### CODE REVIEW

File: accounts/models.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        verbose_name_plural =_('users')
+    
+    def __str__(self):
+        return self.email
+
+    def get_full_name(self):
+        full_name = f'{self.first_name}  {self.last_name}'
+        return full_name.strip()
+
+    def send_email(self, subject, message, from_email=None, **kwargs):
```
Comment (evilcudmovement) [inline accounts/models.py:35]: Should not be here. Please move to celery.

File: banking/models.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+from accounts.models import *
+
+def uuid_str():
+    return str(uuid.uuid4())
+
+
+def get_default_account():
+    """"""
+    num = "".join(random.choices(string.digits, k=12))
+    if num not in Account.objects.all().values_list("account_number", flat=True):
```
Comment (evilcudmovement) [inline banking/models.py:15]: Why not use .exists()?

File: banking/models.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+def get_txn_id():
+    """"""
+    txn_id = "".join(random.choices(string.digits, k=12))
+    if "txn" + txn_id not in Transaction.objects.all().values_list("transaction_id", flat=True):
+        return "txn" + str(txn_id)
+    get_txn_id()
+
+
+class CreatedUpdatedInfo(models.Model):
```
Comment (evilcudmovement) [inline banking/models.py:28]: Rename to BaseModelClass

File: banking/models.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    id  = models.CharField(primary_key=True, default=uuid_str, max_length=100)
+    user = models.OneToOneField(
+        User,
+        related_name="user_customer",
+        on_delete=models.PROTECT,
+        null=True,
+    )
+    customer_id = models.IntegerField(unique=True, editable=False)
+    date_of_birth = models.DateField()
+    pan_card = models.CharField(max_length=10, null=True, blank=True)
```
Comment (evilcudmovement) [inline banking/models.py:66]: rename to pan_card_number and aadhar_card_number

File: banking/models.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+
+class Address(CreatedUpdatedInfo):
+    """
+    """
+    id  = models.CharField(primary_key=True, default=uuid_str, max_length=100)
+    house_no = models.CharField(max_length=10)
+    address_line1 = models.CharField(max_length=200 )
+    address_line2 = models.CharField(max_length=200, null=True, blank=True)
+    city = models.CharField(max_length=50)
```
Comment (evilcudmovement) [inline banking/models.py:85]: city, state, country should be fk to a master table for these

File: banking/models.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    state = models.CharField(max_length=50)
+    county = models.CharField(max_length=50)
+    customer = models.ForeignKey(Customer, on_delete=models.CASCADE, related_name="customer_address")
+    def __str__(self):
+        return str(self.address_line1 +"_"+self.city)
```
Comment (evilcudmovement) [inline banking/models.py:90]: Why are you using str conversion here, also instead of using +, use string formatting

File: banking/models.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        Customer,
+        on_delete=models.SET_NULL,
+        related_name="customer_account",
+        null=True,
+    )
+    account_type = models.CharField(max_length=2, choices=ACCOUNT_CHOICES)
+    status = models.CharField(max_length=2, choices=ACCOUNT_STATUS_CHOICES)
+
+    def __str__(self):
+        return str(str(self.account_number) + "-" + str(self.customer.user.get_full_name()))
```
Comment (evilcudmovement) [inline banking/models.py:142]: use string formatting instead

File: banking/models.py [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+    id  = models.CharField(primary_key=True, default=uuid_str, max_length=100)
+    account_number = models.IntegerField()
+    ifsc = models.CharField(max_length=11)
+    contact_number = models.CharField(max_length=16)
+    customer = models.ForeignKey(
+        Customer, related_name="beneficiary", on_delete=models.PROTECT
+    )
+
+    def __str__(self):
+        return str(self.name + "_" + self.customer.user.get_full_name)
```
Comment (evilcudmovement) [inline banking/models.py:210]: Use stirng formatting

File: banking/models.py [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    def __str__(self):
+        return str(self.user.get_full_name)
+
+
+class Address(CreatedUpdatedInfo):
+    """
+    """
+    id  = models.CharField(primary_key=True, default=uuid_str, max_length=100)
+    house_no = models.CharField(max_length=10)
+    address_line1 = models.CharField(max_length=200 )
```
Comment (evilcudmovement) [inline banking/models.py:83]: rename as address_line_one, address_line_two 


Remove house_no

File: banking/models.py [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+
+    def __str__(self):
+        return str(self.transaction_id)
+
+
+class Beneficiary(CreatedUpdatedInfo):
+    """"""
+    id  = models.CharField(primary_key=True, default=uuid_str, max_length=100)
+    account_number = models.IntegerField()
+    ifsc = models.CharField(max_length=11)
```
[the remark this answers is not in the document]
Comment (deependra-default) [reply banking/models.py:203]: Beneficiary could be outside of available branches.

File: accounts/models.py (hunk 1 above)
Comment (evilcudmovement) [reply accounts/models.py:35]: Then remove the function for now

Comment (evilcudmovement) [reply accounts/models.py:35]: Don't keep things lying around. The PR reviewer doesn't know what you intend to do in the future, they can only evaluate for what they are seeing

**Pull request merged**